### DISCUSSION TITLE
Controller react-select is not focused when there is an error

### DISCUSSION THREAD

**Discussion by raulalonso-aircury** [react-hook-form/react-hook-form#10320] [Q&A] [answered]
Hello, I have the following codesandbox where I recreate the error:

https://codesandbox.io/s/controlledreactselectnotfocus-jsls4w

The problem is that, although there is a ref inside my react-select coming from its controller, the field does not get the focus when an error appear.  I'm quite new with react-hook-form so I don't know if I'm missing something, thank you!

**Comment by leapful** [comment, marked as answer]
@raulalonso-aircury 

react-select does not use native HTML Input element as a UI presentation. The select box is rendered as a custom element with its own focus behavior therefore `inputRef.focus()` will do nothing on focus.

Prop `ref` should be used to register internal focus method of Select component instance.

Working demo: https://codesandbox.io/s/controlledreactselectnotfocus-forked-h92hrj?file=/src/ControlledSingleSelectInput.js

**Reply by raulalonso-aircury** [reply]
Hi @leapful do you know how to do the same but for a react DatePicker?  Thnk you!

**Reply by leapful** [reply]
@raulalonso-aircury 

If you are referring to lib `Hacker0x01/react-datepicker` then yes, the component from that library does not have `inputRef`, only `ref` is available and DOM element should be access via

```
<DatePicker ref={(picker) => {
   picker.input.focus();
}} >
```

**Reply by raulalonso-aircury** [reply]
Thank you! Yes, that was exactly what I was referring to!
